src/vs/platform/keybinding/browser/keybindingServiceImpl.ts

```diff
@@ -182,7 +183,7 @@ export abstract class KeybindingService implements IKeybindingService {
 				e.preventDefault();
 			}
 			let commandId = resolveResult.commandId.replace(/^\^/, '');
-			this._commandService.executeCommand(commandId, resolveResult.commandArgs || {}).done(undefined, err => {
+			this._commandService.executeCommand(commandId, objects.assign({ from: 'keybinding' }, resolveResult.commandArgs || {})).done(undefined, err => {
```

Comment:
:-1: for multiple reasons. For once, the command args can be `undefined`, `null`, or a primitive values. I don't think object.assign is happy in that case. Also, this changes the argument the command is receiving which potentially invalidates its execution.